File: scala/functions/string_manipulation.scala
object StringManipulation {
  def main(args: Array[String]): Unit = {
    val text = "  Hello, World!  "

    println(s"Original: '$text'")
    println(s"Upper: '${text.toUpperCase}'")
    println(s"Lower: '${text.toLowerCase}'")
    println(s"Trim: '${text.trim}'")
    println(s"Replace: '${text.replace("World", "Scala")}'")
    println(s"Split: ${text.trim.split(", ").mkString("[", ", ", "]")}")
    println(s"Contains: ${text.contains("World")}")
    println(s"StartsWith: ${text.trim.startsWith("Hello")}")
    println(s"EndsWith: ${text.trim.endsWith("!")}")
    println(s"Length: ${text.length}")
    println(s"Substring: '${text.trim.substring(0, 5)}'")
    println(s"Repeat: '${"-" * 20}'")
  }
}

File: java/functions/StringManipulation.java
public class StringManipulation {
    public static void main(String[] args) {
        String text = "  Hello, World!  ";

        System.out.println("Original: '" + text + "'");
        System.out.println("Upper: '" + text.toUpperCase() + "'");
        System.out.println("Lower: '" + text.toLowerCase() + "'");
        System.out.println("Trim: '" + text.trim() + "'");
        System.out.println("Replace: '" + text.replace("World", "Java") + "'");
        System.out.println("StartsWith: " + text.trim().startsWith("Hello"));
        System.out.println("EndsWith: " + text.trim().endsWith("!"));
        System.out.println("Length: " + text.trim().length());
        System.out.println("Substring: '" + text.trim().substring(0, 5) + "'");
    }
}
